applications/StructuralMechanicsApplication/custom_utilities/structural_mechanics_element_utilities.cpp

```diff
@@ -41,6 +41,81 @@ bool ComputeLumpedMassMatrix(
     return false;
 }
 
+bool HasRayleighDamping(
+    const Properties& rProperites,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    return (GetRayleighAlpha(rProperites, rCurrentProcessInfo) > 0.0 ||
+            GetRayleighBeta(rProperites, rCurrentProcessInfo) > 0.0);
+}
+
+double GetRayleighAlpha(
+    const Properties& rProperites,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // giving the locally defined setting (through Properties) priority
+    // over the globally defined one (through ProcessInfo)
+    if (rProperites.Has(RAYLEIGH_ALPHA)) {
+        return rProperites[RAYLEIGH_ALPHA];
+    }
+    else if (rCurrentProcessInfo.Has(RAYLEIGH_ALPHA)) {
+        return rCurrentProcessInfo[RAYLEIGH_ALPHA];
+    }
+
+    return 0.0;
+}
+
+double GetRayleighBeta(
+    const Properties& rProperites,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // giving the locally defined setting (through Properties) priority
+    // over the globally defined one (through ProcessInfo)
+    if (rProperites.Has(RAYLEIGH_BETA)) {
+        return rProperites[RAYLEIGH_BETA];
+    }
+    else if (rCurrentProcessInfo.Has(RAYLEIGH_BETA)) {
+        return rCurrentProcessInfo[RAYLEIGH_BETA];
+    }
+
+    return 0.0;
+}
+
+void CalculateRayleighDampingMatrix(
+    Element& rElement,
+    Element::MatrixType& rDampingMatrix,
+    /*const*/ProcessInfo& rCurrentProcessInfo,
+    const std::size_t MatrixSize)
+{
+    KRATOS_TRY;
+
+    // 1.-Resizing if needed
+    if (rDampingMatrix.size1() != MatrixSize || rDampingMatrix.size2() != MatrixSize) {
+        rDampingMatrix.resize( MatrixSize, MatrixSize, false );
+    }
+    noalias(rDampingMatrix) = ZeroMatrix(MatrixSize, MatrixSize);
+
+    if (!HasRayleighDamping(rElement.GetProperties(), rCurrentProcessInfo)) {
+        // Do nothing if no rayleigh-damping is specified
+        return;
```

Comment:
For those interested:
With this check I managed to reduce the time a simple simulation with a few beam-elements takes by a factor of over 2!
I was not aware it is such a big difference, but good for us :D